lib/Doctrine/DBAL/Schema/OracleSchemaManager.php

```diff
@@ -36,6 +36,52 @@ class OracleSchemaManager extends AbstractSchemaManager
     /**
      * {@inheritdoc}
      */
+    public function listTables()
+    {
+        $currentDatabase = $this->_conn->getDatabase();
+
+        $tableNames = $this->listTableNames();
+
+        // Get all column definitions in one database call.
+        $allColumns = $this->_conn->fetchAll($this->_platform->getListTableColumnsSQL(null, $currentDatabase));
+
+        // Get all foreign keys definitions in one database call.
+        $allForeignKeys = $this->_conn->fetchAll($this->_platform->getListForeignKeysSQL(null, $currentDatabase));
+
+        // Get all indexes definitions in one database call.
+        $allIndexes = $this->_conn->fetchAll($this->_platform->getListTableIndexesSQL(null, $currentDatabase));
+
+        $tables = [];
+        foreach ($tableNames as $tableName) {
+            $unquotedTableName = rtrim(ltrim($tableName, '"'), '"');
```

Comment:
This way, this code depends on the implementation details of the `OraclePlatform::listTableNames()` — only `OraclePlatform` returns table names quoted (not sure why) while the others don't.

> Since listTableNames() does all the checks/validations upfront, I'd prefer to keep this as is.

That's a valid point which brings up the question: if `AbstractSchemaManager::_getPortableTableDefinition()` returns quoted or non-quoted identifiers based on the platform, how is the behavior of `filterSchemaAssetsExpression` (I assume this is what was meant by checks/validations above) is expected to be portable? Or it's not? /cc @Ocramius